pennylane/templates/subroutines/trotter.py

```diff
@@ -490,3 +492,340 @@ def compute_decomposition(*args, **kwargs):
+        n (int): an integer representing the number of Trotter steps to perform
+        order (int): an integer (:math:`m`) representing the order of the approximation (must be 1 or even)
+        reverse (bool): if true, reverse the order of the operations queued by :code:`qfunc`
+        name (str): an optional name for the instance
+        **non_trainable_kwargs (dict): non-trainable keyword arguments of the first-order expansion function
+
+    Raises:
+        ValueError: A qfunc must be provided to be trotterized.
+
+    **Example**
+
+    .. code-block:: python3
+
+        from pennylane.templates.subroutines.trotter import TrotterizedQfunc
+
+        def first_order_expansion(time, theta, phi, wires=[0, 1, 2], flip=False):
+            "This is the first order expansion (U_1)."
+            qml.RX(time*theta, wires[0])
+            qml.RY(time*phi, wires[1])
+            if flip:
+                qml.CNOT(wires=wires[:2])
+
+        @qml.qnode(qml.device("default.qubit"))
+        def my_circuit(time, angles, num_trotter_steps):
+            TrotterizedQfunc(
+                time,
+                *angles,
+                qfunc=first_order_expansion,
+                n=num_trotter_steps,
+                order=2,
+                wires=['a', 'b', 'c'],
+                flip=True,
+            )
+            return qml.state()
+
+    We can visualize the circuit to see the Suzuki-Trotter product formula being applied:
+
+        >>> time = 0.1
+        >>> angles = (0.12, -3.45)
+        >>> print(qml.draw(my_circuit, level=3)(time, angles, num_trotter_steps=1))
+        a: ──RX(0.01)──╭●─╭●──RX(0.01)──┤  State
+        b: ──RY(-0.17)─╰X─╰X──RY(-0.17)─┤  State
+        >>>
+        >>> print(qml.draw(my_circuit, level=3)(time, angles, num_trotter_steps=3))
+        a: ──RX(0.00)──╭●─╭●──RX(0.00)───RX(0.00)──╭●─╭●──RX(0.00)───RX(0.00)──╭●─╭●──RX(0.00)──┤  State
+        b: ──RY(-0.06)─╰X─╰X──RY(-0.06)──RY(-0.06)─╰X─╰X──RY(-0.06)──RY(-0.06)─╰X─╰X──RY(-0.06)─┤  State
+
+    """
+
+    def __init__(
+        self,
+        time,
+        *trainable_args,
+        qfunc=None,
+        n=1,
+        order=2,
+        reverse=False,
+        name=None,
+        id=None,
+        **non_trainable_kwargs,
+    ):
+        # This class requires the input function (qfunc) has a very specific
+        # signature: qfunc(time, arg1, ..., arg_n, wires=[...], kwarg_1, ..., kwarg_n)
+
+        if qfunc is None:
+            raise ValueError("The qfunc must be provided to be trotterized.")
+
+        if order <= 0 or order != 1 and order % 2 != 0:
+            raise ValueError(f"The order must be 1 or a positive even integer, got {order}.")
+
+        try:
+            wires = non_trainable_kwargs.pop("wires")
+        except KeyError:
+            wires = trainable_args[-1]
+            trainable_args = trainable_args[:-1]  # exclude the wires from the args
+
+        self._hyperparameters = non_trainable_kwargs
```

Comment:
Could we have "non_trainable_args" as a key with a dictionary value instead? This might help have a clearer distinction between operator kwargs and qfunc kwargs.

Reply:
Discussed offline and decided to keep the non_trainable_args as is.